File: api/OpenSkull.Api/DTO/Game.cs
namespace OpenSkull.Api.DTO;


public enum CardType {
  Flower,
  Skull
}

public enum CardState {
  InPlay,
  Discarded
}

public record struct Card {
  public Guid Id { get; set; }
  public CardType Type { get; set; }
  public CardState State { get; set; }
}

public record struct Game {
  public Guid[] PlayerIds { get; set; }
  public Card[][] PlayerCards { get; set; }
  public int ActivePlayerIndex { get; set; }
  public List<List<Guid>[]> RoundPlayerCardIds { get; set; }
  public List<int[]> RoundBids { get; set; }
  public List<List<int>> RoundRevealedCardPlayerIndexes { get; set; }
  public List<int> RoundWinPlayerIndexes { get; set; }
  public bool GameComplete { get; set; }
}

File: webapp/src/models/Game.ts
export enum CardType {
  Flower,
  Skull
}

export enum CardState {
  InPlay,
  Discarded
}


export enum TurnAction {
  Card,
  Bid,
  Flip
}

export enum RoundPhase {
  PlayFirstCards,
  PlayCards,
  Bidding,
  Flipping
}


export interface Card {
    id: string;
    type: CardType;
    state: CardState;
}

export interface PublicGame 
{
  id: string;
  activePlayerIndex: number;
  playerCount: number;
  playerNicknames: string[];
  playerCardStartingCount: number;
  roundNumber: number;
  currentCountPlayerCardsAvailable: number[];
  currentBids: number[];
  currentRoundPhase: RoundPhase;
  roundCountPlayerCardsPlayed: number[][];
  roundPlayerCardsRevealed: CardType[][][];
  roundWinners: number[];
  gameComplete: boolean;
  lastUpdated: Date;
}

export interface PlayerGame extends PublicGame
{
  playerId: string;
  playerIndex: number;
  playerCards: Card[];
  playerRoundCardIdsPlayed: string[][];
}
